src/libsync/discovery.cpp

```diff
@@ -85,8 +85,7 @@ void ProcessDirectoryJob::process()
     for (auto &e : _localNormalQueryEntries) {
         // Remove the virtual file suffix
         auto name = e.name;
-        if (hasVirtualFileSuffix(name)) {
-            e.isVirtualFile = true;
+        if (e.isVirtualFile && isVfsWithSuffix()) {
```

Comment:
maybe change chopVirtualFileSuffix to return bool, so you can do  
`if (e.isVirtualFile && chopVirtualFileSuffix(name))`

since this pattern is repeated a lot.

Likewise the addVirtualFileSuffix could be a no-op if the mode does not have the suffix

Reply:
I thought the same thing and looked at it again. I prefer the explicitness of checking for suffix-vfs. I feel I'd need to rename to `maybeChopSuffix` and `maybeAddSuffix` otherwise and that makes the code less clear.